hail/src/main/scala/is/hail/rvd/AbstractRVDSpec.scala

```diff
@@ -189,16 +189,23 @@ object AbstractRVDSpec {
     val partitioner = specLeft.partitioner
 
     val extendedNewPartitioner = newPartitioner.map(_.extendKey(partitioner.kType))
-    val tmpPartitioner = extendedNewPartitioner match {
-      case Some(np) => np.intersect(partitioner)
-      case None => partitioner
+    val (parts, tmpPartitioner) = extendedNewPartitioner match {
+      case Some(np) =>
+        val tmpPart = np.intersect(partitioner)
+        assert(specLeft.key.nonEmpty)
+        val p = tmpPart.rangeBounds.map { b => specLeft.partFiles(partitioner.lowerBoundInterval(b)) }
+        (p, tmpPart)
+      case None =>
+        // need to remove partitions with degenerate intervals
+        // these partitions are necessarily empty
+        val iOrd = partitioner.kord.intervalEndpointOrdering
+        val includedIndices = (0 until partitioner.numPartitions).filter { i =>
+          val rb = partitioner.rangeBounds(i)
+          rb.isDisjointFrom(iOrd, rb)
+        }.toArray
+        (includedIndices.map(specLeft.partFiles), partitioner.copy(rangeBounds = includedIndices.map(partitioner.rangeBounds)))
```

Comment:
I actually think the None case could just be `(specLeft.partFiles, partitioner)`, but I'm okay with dropping degenerate partitions too.

Reply:
Oh! Yes, this is better. The root problem is that the lower bound is not meaningful, but this is fine.